lib/pavilion/plugins/commands/clean.py

```diff
@@ -73,146 +121,20 @@ def run(self, pav_cfg, args):
                 elif args.older_than[1] in ['month', 'months']:
                     cutoff_date = datetime.today() - timedelta(
                         days=30*int(args.older_than[0]))
+
             elif len(args.older_than) == 3:
                 date = ' '.join(args.older_than)
                 try:
                     cutoff_date = datetime.strptime(date, '%b %d %Y')
-                except (TypeError, ValueError):
-                    output.fprint("{} is not a valid date."
-                                  .format(args.older_than),
-                                  file=self.errfile, color=output.RED)
-                    return errno.EINVAL
+                except ValueError as err:
+                    raise ValueError("{} is not a valid date: {}".format(date,
+                                                                        err))
             else:
-                output.fprint(
-                    "Invalid `--older-than` value.", file=self.errfile,
-                    color=output.RED
```

Comment:
Same here.

Reply:
Now, if you move functionality into other functions than `run`, you have to pass up error information somehow. Doing so with an exception is generally reasonable, but you should have an exception specific to the module to use and catch.